taichi/rhi/metal/metal_device.h

```diff
@@ -358,7 +404,20 @@ class MetalDevice final : public GraphicsDevice {
     return *default_sampler_;
   }
 
+  const BufferFormat get_swap_image_format() {
+    return swap_chain_image_format_;
+  }
+
+  MTLFunction_id get_mtl_function(MTLLibrary_id mtl_lib,
+                                  const std::string &func_name) const;
+  MTLLibrary_id get_mtl_library(const std::string &source) const;
+
  private:
+  const BufferFormat swap_chain_image_format_{BufferFormat::bgra8};
+
+  const std::string frag_function_name = "frag_function";
```

Comment:
Put it outside the class and use `constexpr`, also constants uses a different formatting.
Afterall `const` just means non-mutable. `constexpr` means actual compile time constants

Should be
```c++
constexpr std::string kMTLFragFunctionName = "frag_function";
```